### PR TITLE
[hotfix]different overflow status lead to communication stuck.

### PR DESCRIPTION
**Pull request by YuliangLiu0306**

### CODE REVIEW

File: colossalai/amp/naive_amp/_fp16_optimizer.py [hunk 1]
```diff
@@ -258,24 +258,25 @@ def step(self):
...
-        # Update params from main params.
-        self._update_fp16_param_from_fp32_param()
+            # Update params from main params.
+            self._update_fp16_param_from_fp32_param()
 
-        # Successful update.
-        return True, grad_norm
+            # Successful update.
+            return True, grad_norm
+        else:
```
Comment (feifeibear) [inline colossalai/amp/naive_amp/_fp16_optimizer.py:278]: why clip grad if overflow?

Comment (YuliangLiu0306) [reply colossalai/amp/naive_amp/_fp16_optimizer.py:278]: Different pipeline ranks may have different overflow status, if the overflow do not join the clip grad norm, the all reduce process will be stuck.

Comment (YuliangLiu0306) [reply colossalai/amp/naive_amp/_fp16_optimizer.py:278]: Another solution add an extra communication for overflow status, then we could skip the clip grad and step if some ranks overflowed.

Comment (feifeibear) [reply colossalai/amp/naive_amp/_fp16_optimizer.py:278]: I see!

**Review by feifeibear — APPROVED** [review]

**Pull request merged**